src/dash-table/derived/cell/wrappers.tsx

```diff
@@ -19,40 +19,76 @@ class Wrappers {
 
     }
 
+    partialGet = memoizeOne((
+        columns: VisibleColumns,
+        data: Data,
+        _offset: IViewportOffset
+    ) => R.addIndex<Datum, JSX.Element[]>(R.map)(
+        (_, rowIndex) => R.addIndex<IVisibleColumn, JSX.Element>(R.map)(
+            (column, columnIndex) => this.getWrapper(
+                false,
+                false,
+                rowIndex,
+                columnIndex,
+                column
+            ), columns), data));
+
     /**
      * Returns the wrapper for each cell in the table.
      */
     get = memoizeOne((
-        activeCell: ICellCoordinates | undefined,
-        columns: VisibleColumns,
-        data: Data,
+        wrappers: JSX.Element[][],
         offset: IViewportOffset,
+        activeCell: ICellCoordinates | undefined,
         selectedCells: SelectedCells
-    ) => R.addIndex<Datum, JSX.Element[]>(R.map)(
-        (_, rowIndex) => R.addIndex<IVisibleColumn, JSX.Element>(R.map)(
-            (column, columnIndex) => {
-                const active = isActiveCell(activeCell, rowIndex + offset.rows, columnIndex + offset.columns);
-                const selected = isSelectedCell(selectedCells, rowIndex + offset.rows, columnIndex + offset.columns);
-                const isDropdown = column.presentation === Presentation.Dropdown;
-                const classes =
-                    'dash-cell' +
-                    ` column-${columnIndex}` +
-                    (active ? ' focused' : '') +
-                    (selected ? ' cell--selected' : '') +
-                    (isDropdown ? ' dropdown' : '');
+    ) => {
+        wrappers = shallowClone(wrappers);
+
+        R.forEach(({ row: i, column: j }) => {
+            i -= offset.rows;
+            j -= offset.columns;
 
-                return this.wrapper.get(rowIndex, columnIndex)(
-                    active,
-                    classes,
-                    columnIndex,
-                    column.id,
-                    rowIndex,
-                    this.handlers(Handler.Enter, rowIndex, columnIndex),
-                    this.handlers(Handler.Leave, rowIndex, columnIndex),
-                    this.handlers(Handler.Move, rowIndex, columnIndex)
-                );
-            }, columns
-        ), data));
+            if (i < 0 || j < 0 || wrappers.length <= i || wrappers[i].length <= j) {
```

Comment:
Now looping on `selected_cells` instead of data/column. Need to check boundaries.